src/main/java/project/onlinebookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,24 @@
+package project.onlinebookstore.repository.book;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import project.onlinebookstore.model.Book;
+import project.onlinebookstore.repository.SpecificationProvider;
+import project.onlinebookstore.repository.SpecificationProviderManager;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> specificationProviderList;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return specificationProviderList
+                .stream()
+                .filter(provider -> provider.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Can't find correct "
```

Comment:
throw more suitable NoSuchElementException

Reply:
not fixed